### DISCUSSION TITLE
what's the point of context in redis client?

### DISCUSSION THREAD

**Discussion by zYasser** [redis/go-redis#3084] [Q&A] [answered]
Hello, 
i have been reading the quick start and i saw that we used Context for create a data

```
var ctx = context.Background()

func ExampleClient() {
    rdb := redis.NewClient(&redis.Options{
        Addr:     "localhost:6379",
        Password: "", // no password set
        DB:       0,  // use default DB
    })

    err := rdb.Set(ctx, "key", "value", 0).Err()
    if err != nil {
        panic(err)
    }

```
what's the point of it ?, why we don't just pass the key and value ?
thank for help

**Comment by RPGillespie6** [comment, marked as answer]
For example, you might be connecting to redis in an HTTP handler, and if the HTTP client terminates the connection, there's no point in continuing on with some redis operation. The context lets redis client know that the operation has been cancelled.
